src/redux/features/authDetailsSlice.ts

```diff
@@ -0,0 +1,32 @@
+import { Token } from '@/types/common'
+import { createSlice } from '@reduxjs/toolkit'
+import { RootState } from '@/redux/store'
+
+interface IInitialState {
+  tokenPayload: Token | {}
+  internalUserId: string
+}
+
+const initialState: IInitialState = {
+  tokenPayload: {},
+  internalUserId: '',
+}
+
+const authDetailsSlice = createSlice({
+  name: 'authDetails',
+  initialState,
+  reducers: {
+    setTokenPayload: (state, action) => {
+      state.tokenPayload = action.payload
+    },
+    setInternalUserId: (state, action) => {
+      state.internalUserId = action.payload
+    },
```

Comment:
Please give types to the `action` as we have started typing the actions now after suggestion from SM dai. You can have take reference from other slices.